website/docs/Examples/Integrate - Spark.md

```diff
@@ -0,0 +1,112 @@
+- `index_col` is the column name to use as the index, default is None.
+- `default_index_type` is the default index type, default is "distributed-sequence". More info about default index type could be found on Spark official [documentation](https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/options.html#default-index-type)
+
+Here is an example code snippet for Spark Data:
+
+```python
+import pandas as pd
+from flaml.automl.spark.utils import to_pandas_on_spark
+# Creating a dictionary
+data = {"Square_Feet": [800, 1200, 1800, 1500, 850],
+      "Age_Years": [20, 15, 10, 7, 25],
+      "Price": [100000, 200000, 300000, 240000, 120000]}
+
+# Creating a pandas DataFrame
+dataframe = pd.DataFrame(data)
+label = "Price"
+
+# Convert to pandas-on-spark dataframe
+psdf = to_pandas_on_spark(dataframe)
+```
+
+To use Spark ML models you need to format your data appropriately. Specifically, use [`VectorAssembler`](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.feature.VectorAssembler.html) to merge all feature columns into a single vector column.
+
+Here is an example of how to use it:
+```python
+from pyspark.ml.feature import VectorAssembler
+columns = psdf.columns
+feature_cols = [col for col in columns if col != label]
+featurizer = VectorAssembler(inputCols=feature_cols, outputCol="features")
+psdf = featurizer.transform(psdf.to_spark(index_col="index"))["index", "features"]
+```
+
+Later in conducting the experiment, use your pandas-on-spark data like non-spark data and pass them using `X_train, y_train` or `dataframe, label`.
+
+### Estimators
+#### Model List
+- `lgbm_spark`: The class for fine-tuning Spark version LightGBM models, using [SynapseML](https://microsoft.github.io/SynapseML/docs/features/lightgbm/about/) API.
+
+#### Usage
+First, prepare your data in the required format as described in the previous section.
+
+By including the models you intend to try in the `estimators_list` argument to `flaml.automl`, FLAML will start trying configurations for these models. If your input is Spark data, FLAML will also use estimators with the `_spark` postfix by default, even if you haven't specified them.
+
+Here is an example code snippet using SparkML models in AutoML:
+
+```python
+import flaml
+# prepare your data in pandas-on-spark format as we previously mentioned
+
+automl = flaml.AutoML()
+settings = {
+    "time_budget": 30,
+    "metric": "r2",
+    "estimator_list": ["lgbm_spark"],  # this setting is optional
+    "task": "regression",
+}
+
+automl.fit(
+    dataframe=psdf,
+    label=label,
+    **settings,
+)
+```
+
+## Parallel Spark Jobs
+In [parallel tuning](../Use-Cases/Task-Oriented-AutoML#parallel-tuning), you can use Spark as the parallel backend by setting `use_spark` to `true` in both Hyperparameter Tuning and AutoML. FLAML will dispatch your job to the distributed Spark backend using [`joblib-spark`](https://github.com/joblib/joblib-spark).
```

Comment:
There are two use cases involving "parallel tuning", task-oriented-automl and tune-user-defined-functions.

Reply:
sure, I'll try to reference both by mentioning from "AutoML" and "Hyperparameter Tuning"